src/08_ecosystem/00_chapter.md

```diff
@@ -0,0 +1,51 @@
+# The Async Ecosystem
+Most Rust asynchronous projects will need third-party crates to function.
+The `futures` crate contains traits and functions useful for writing async code.
+In addition, there are several crates that provide async runtimes,
+which are used to run asynchronous functions from synchronous code.
+This section introduces the crates most commonly used in async projects.
+
+## Asynchronous Runtimes
+Async runtimes are libraries that allow async functions to be run from sync functions, including the `main` function.
+Runtimes usually bundle together a *reactor* with one or more *executors*.
+Reactors can be thought of as event queues.
+When an event occurs, such as readable data becoming available in a socket,
+the reactor is responsible for adding a task to the queue to process this event.
+An executor handles execution of any tasks in the event queue.
+It keeps track of running and suspended tasks, polls futures to completion, and wakes tasks when they can make progress.
+The word "executor" is frequently used interchangeably with "runtime".
+
+## Choosing a Runtime
+There is no asynchronous runtime in the standard library, and none are officially recommended.
+The following crates provide popular runtimes.
+
+- [futures](https://docs.rs/futures/): The `futures` crate is not an async ecosystem like other crates in this list,
+but it provides useful traits like `Stream`, `Sink`, `AsyncRead`, and `AsyncWrite`, and utilities such as combinators.
+These utilities and traits may eventually become part of the standard library.
+`futures` has an executor, `futures-executor`, but a common choice is to use utilities from `futures` with an
+executor from another crate.
+- [Tokio](https://docs.rs/tokio/): A popular async ecosystem with HTTP, gRPC, and tracing frameworks.
+- [async-std](https://docs.rs/async-std/): A crate that provides asynchronous counterparts to standard library components.
+- [smol](https://docs.rs/smol/): A small, simplified async runtime.
+Provides the `Async` trait that can be used to wrap structs like `UnixStream` or `TcpListener`.
+- [fuchsia-async](https://fuchsia.googlesource.com/fuchsia/+/master/src/lib/fuchsia-async/): An executor for use in the Fuchsia OS.
+
+Not all async runtimes are compatible with each other.
+Notably, `Tokio` uses the `mio` reactor and defines its own versions of async I/O traits, including `AsyncRead` and `AsyncWrite`.
+It's not compatible with `async-std` and `smol`, which rely on the [`async-executor` crate](https://docs.rs/async-executor).
+The [`async_compat` crate](https://docs.rs/async_compat) can be used for compatibility between `Tokio` and other runtimes.
+
+If possible, libraries should not depend on specific runtimes.
```

Comment:
Perhaps clarify where it is necessary:

Libraries that expose async APIs generally should not depend on a specific executor or reactor, unless:

- It needs to spawn its own tasks
- Defines its own async IO or timer futures

Reply:
Can you explain more what you mean about defining your own async IO or timer futures? and what's an example of a situation where it's better to spawn tasks in the library than in a binary?

Reply:
Resolving as it doesn't sound like additional changes are necessary here.